src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -615,6 +649,99 @@ const getSshInfo = async (frameworkName) => {
   throw createError('Not Found', 'NoJobSshInfoError', `SSH info of job ${frameworkName} is not found.`);
 };
 
+const generateExitDiagnostics = (diag) => {
+  if (_.isEmpty(diag)) {
+    return null;
+  }
+
+  const exitDiagnostics = {
+    diagnosticsSummary: diag,
+    runtime: null,
+    launcher: diag,
+  };
+  const regex = /matched: (.*)/;
+  const matches = diag.match(regex);
+
+  // No container info here
+  if (matches === null || matches.length < 2) {
+    return exitDiagnostics;
+  }
+
+  let podCompletionStatus = null;
+  try {
+    podCompletionStatus = JSON.parse(matches[1]);
+  } catch (error) {
+    logger.warn('Get diagnostics info failed', error);
+    return exitDiagnostics;
+  }
+
+  const summmaryInfo = diag.substring(0, matches.index + 'matched:'.length);
```

Comment:
>matches.index [](start = 41, length = 13)

What is this matches.index? multiple values?

Reply:
index mean the postion where patten is found. refer: [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match)

---
In reply to: [323575627](https://github.com/microsoft/pai/pull/3575#discussion_r323575627) [](ancestors = 323575627)